x-pack/plugins/security_solution/server/lib/detection_engine/rules/bulk_action_edit.ts

```diff
@@ -73,7 +96,17 @@ export const applyBulkActionEditToRule = (
       );
       invariant(action.value.length !== 0, "Index patterns can't be overwritten with empty list");
 
+      if (
+        (isEqlParams(rule.params) ||
+          isQueryParams(rule.params) ||
+          isThresholdParams(rule.params) ||
+          isThreatParams(rule.params)) &&
+        action.overwriteDataViews
```

Comment:
nit: Might be nice to pull this out into a function since it's used three times here (+help with readability with a nice function name as well).